test/integration/package_controller_test.rb

```diff
@@ -0,0 +1,55 @@
+require File.expand_path('../../test_helper', __FILE__)
+require 'minitest/mock'
+require 'digest'
+require 'fileutils'
+
+class PackageControllerTest < ActionDispatch::IntegrationTest
+  FIREFOX_THUMBNAIL = File.join(Rails.root, 'public', 'images', 'thumbnails', 'MozillaFirefox.png')
+  PKG_4PANE_THUMBNAIL = File.join(Rails.root, 'public', 'images', 'thumbnails', '4pane.png')
+
+  def test_thumbnail_unknown_package_returns_default_asset
+    stub = proc { |arg| arg == FIREFOX_THUMBNAIL ? false : File.exist?(arg) }
```

Comment:
what about writing this like:

```ruby
stub = proc { |arg| arg != FIREFOX_THUMBNAIL && File.exist?(arg) }

Reply:
I got rid of the stubs completely as I was not sure if it was supported to nest them (stubbing File.exists and then using it inside the stub definition).

Reply:
But I am only suggesting writing this line differently